internal/config/env.go

```diff
@@ -0,0 +1,50 @@
+package config
+
+import (
+	"log"
+	"strconv"
+
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/gobuffalo/envy"
+)
+
+// ParseEnvironment loads a sibling `.env` file then looks through all environment
+// variables to set global configuration.
+func ParseEnvironment() error {
+	envy.Load()
+	azureEnv, _ := azure.EnvironmentFromName("AzurePublicCloud") // shouldn't fail
+
+	// these must be provided by environment
+	var err1, err2, err3, err4 error
+	clientID, err1 = envy.MustGet("AZURE_CLIENT_ID")
+	clientSecret, err2 = envy.MustGet("AZURE_CLIENT_SECRET")
```

Comment:
I'm surprised these are required. I thought there was some ability to use Device flow?

Reply:
Tried to get fancy with error handling but will change to handle each and exclude client secret when device flow is used.

I see now why a framework for configuration is helpful, it's hard to handle precedence and relationships between different config sources smoothly.